src/tests/Microsoft.Diagnostics.NETCore.Client/TestRunner.cs

```diff
@@ -20,6 +20,9 @@ public class TestRunner : IDisposable
         private ProcessStartInfo startInfo;
         private ITestOutputHelper outputHelper;
 
+        private CancellationTokenRegistration ctsRegistration;
```

Comment:
You can ignore caching the registration. Disposing the cts will clear all registrations, clean the timer, and some other handles. Given that their lifetime is the same, it's enough to just get the cts as a field.